### ISSUE TITLE
I see extension .javajava on files

### ISSUE THREAD

**Issue by kuzminva** [TouK/sputnik#125]
10:37:06.704 [main] INFO  p.t.s.p.f.CollectorBugReporter - FindBugs audit finished
10:37:06.707 [main] WARN  p.t.s.p.f.CollectorBugReporter - Analysis error File not found: filesystem:/home/jenkins/components/ingestion/service/src/com/product/indexing/Indexer.javajava

It seems like a bug or I can change some option?

**Comment by kuzminva** [comment]
I guess that javajava extension is just a typo in log message. Main problem, gerrit returns filename as components/ingestion/service/src/com/product/indexing/Indexer.java but jenkins workspace is /home/jenkins/workspace/components/ingestion/service/src/com/product/indexing/Indexer.java

Is there way to configure it?

**Comment by SpOOnman** [comment]
It seems like a bug. Are other files correctly handled by FindBugs? Are other analysers fine with this file?

**Comment by kuzminva** [comment]
The problem is that my projects uses maven + ant. And as I see Sputnik doesn't support ant. I will need to create AntBuildDirLocator for my case.

**Comment by SpOOnman** [comment]
Yes, sputnik does not support ant. It would be great if you can create that!

**Comment by kuzminva** [comment]
I managed to hack Sputnik to make it working with Ant. But I made a change to 1.5.1 changeset. 

Mostly it is:

 public interface Paths {
 
-    public static final String ENTRY_REGEX = ".*src/(main|test)/java/";
+    public static final String ENTRY_REGEX = ".*src/";

-    public static final String SRC_MAIN = "src/main/";
-    public static final String SRC_TEST = "src/test/";
+    public static final String SRC_MAIN = "src/";
+    public static final String SRC_TEST = "test/";

and

     @NotNull
     public String getSourceDir() {
-        return StringUtils.substringBeforeLast(reviewFilename, JAVA).concat(JAVA);
+       return StringUtils.substringBeforeLast(reviewFilename, "com");

AntBuildDirLocator is also about "src" and "test"

+public class AntBuildDirLocator implements BuildDirLocator {
+
+       public List<String> getBuildDirs(Review review) {
+        return Lists.transform(review.getFiles(), new Function<ReviewFile, String>() {
+            @Nullable
+            @Override
+            public String apply(ReviewFile file) {
+                if (file.getReviewFilename().contains("src")) {
+                    return substringBeforeLast(file.getReviewFilename(), "src").concat("build");
+                }
+                if (file.getReviewFilename().contains("test")) {
+                    return substringBeforeLast(file.getReviewFilename(), "test").concat("build");
+                }
+
+                return file.getSourceDir();
+            }
+        });
+    }
+}

In fact, I think that it's better to avoid build-system awareness in a code and use some flexible configuration for Paths (ENTRY_REGEX, SRC_MAIN, SRC_TST, BUILD) and "org/com" things. And some pre-set values for Maven, Gradle. Also, if allow "regex" configuration, it could support mixes like in my project mvn+ant.

**Comment by kuzminva** [comment]
I also thought that for mixed maven + ant I could add post-build 2 jobs with 2 different configs, but in this case it will report 2 times that will be confusing for user. One of the possible solution - support multiple configs and run sputnik internally twice with configs one by one and only then report.

**Comment by SpOOnman** [comment]
@kuzminva I think we should keep Sputnik simple as possible and avoid duplicate runs. You can run it two times with different configurations. As for now comments will duplicate.

In your build are maven sources and ant sources mixed or disjunctive? And why on earth do you have two? :)
